tests/core/featurizers/test_tracker_featurizer.py

```diff
@@ -49,48 +64,2238 @@ def test_convert_labels_to_ids(domain: Domain):
+        state_featurizer, max_history=max_history
+    )
+    interpreter = RegexInterpreter()
+    state_featurizer.prepare_for_training(moodbot_domain, interpreter)
+    actual_features = tracker_featurizer.create_state_features(
+        [tracker], moodbot_domain, interpreter,
+    )
+
+    expected_features = [
+        [
+            {},
+            {
+                INTENT: [moodbot_features["intents"]["greet"]],
+                ACTION_NAME: [moodbot_features["actions"][ACTION_LISTEN_NAME]],
+            },
+            {ACTION_NAME: [moodbot_features["actions"][ACTION_UNLIKELY_INTENT_NAME]]},
+            {ACTION_NAME: [moodbot_features["actions"]["utter_greet"]]},
+            {
+                INTENT: [moodbot_features["intents"]["mood_great"]],
+                ACTION_NAME: [moodbot_features["actions"][ACTION_LISTEN_NAME]],
+            },
+            {ACTION_NAME: [moodbot_features["actions"][ACTION_UNLIKELY_INTENT_NAME]]},
+            {ACTION_NAME: [moodbot_features["actions"]["utter_happy"]]},
+        ],
+    ]
+
+    if max_history is not None:
+        expected_features = [x[-max_history:] for x in expected_features]
+
+    assert actual_features is not None
+    assert len(actual_features) == len(expected_features)
+
+    for actual, expected in zip(actual_features, expected_features):
+        assert compare_featurized_states(actual, expected)
+
+
+@pytest.mark.parametrize("max_history", [None, 2])
+def test_prediction_states_with_intent_max_history_tracker_featurizer(
+    moodbot_tracker: DialogueStateTracker,
+    moodbot_domain: Domain,
+    max_history: Optional[int],
+):
+
+    # IntentMaxHistoryTrackerFeaturizer prediction is only done after
+    # a UserUttered event so remove the last BotUttered and
+    # ActionExecuted events.
+    moodbot_tracker = moodbot_tracker.copy()
+    moodbot_tracker.events.pop()
+    moodbot_tracker.events.pop()
+
+    state_featurizer = IntentTokenizerSingleStateFeaturizer()
+    tracker_featurizer = IntentMaxHistoryTrackerFeaturizer(
+        state_featurizer, max_history=max_history
+    )
+    actual_states = tracker_featurizer.prediction_states(
+        [moodbot_tracker], moodbot_domain,
+    )
+
+    expected_states = [
+        [
+            {},
+            {
+                USER: {INTENT: "greet"},
+                PREVIOUS_ACTION: {ACTION_NAME: ACTION_LISTEN_NAME},
+            },
+            {USER: {INTENT: "greet"}, PREVIOUS_ACTION: {ACTION_NAME: "utter_greet"},},
+            {
+                USER: {INTENT: "mood_unhappy"},
+                PREVIOUS_ACTION: {ACTION_NAME: ACTION_LISTEN_NAME},
+            },
+            {
+                USER: {INTENT: "mood_unhappy"},
+                PREVIOUS_ACTION: {ACTION_NAME: "utter_cheer_up"},
+            },
+            {
+                USER: {INTENT: "mood_unhappy"},
+                PREVIOUS_ACTION: {ACTION_NAME: "utter_did_that_help"},
+            },
+        ]
+    ]
```

Comment:
I don't understand why this is identical to `expected_state` in `test_prediction_states_with_max_history_tracker_featurizer` but missing the last two entries.

Reply:
> IntentMaxHistoryTrackerFeaturizer will filter out the last UserUttered and action_listen events for reasons that are explained here.

Your linked comment explains this. Thanks. 

> This tracker is different than the one that you are mentioning for the equivalent MaxHistoryTrackerFeaturizer test. 

Thanks. I can see that you pop the last two events.